drivers/sensor/sbs_gauge/sbs_gauge.c

```diff
@@ -0,0 +1,272 @@
+		break;
+
+	case SENSOR_CHAN_GAUGE_STATE_OF_CHARGE:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_ASOC,
+					  &data->state_of_charge);
+		if (status < 0) {
+			LOG_ERR("Failed to read state of charge");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_FULL_CAPACITY,
+					  &data->full_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full charge capacity");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_REMAINING_CHARGE_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_REM_CAPACITY,
+					  &data->remaining_charge_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read remaining charge capacity");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_NOM_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_NOM_CAPACITY,
+					  &data->nom_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read nominal available capacity");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_FULL_AVAIL_CAPACITY:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_FULL_CAPACITY,
+					  &data->full_avail_capacity);
+		if (status < 0) {
+			LOG_ERR("Failed to read full available capacity");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_EMPTY:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_AVG_TIME2EMPTY,
+					  &data->time_to_empty);
+		data->time_to_empty = (data->time_to_empty) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read time to empty");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_TIME_TO_FULL:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_AVG_TIME2FULL,
+					  &data->time_to_full);
+		data->time_to_full = (data->time_to_full) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read time to full");
+		}
+		break;
+
+	case SENSOR_CHAN_GAUGE_CYCLE_COUNT:
+		status = sbs_cmd_reg_read(dev,
+					  SBS_GAUGE_CMD_CYCLE_COUNT,
+					  &data->cycle_count);
+		data->cycle_count = (data->cycle_count) & 0x00FF;
+
+		if (status < 0) {
+			LOG_ERR("Failed to read cycle count");
+		}
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return status;
+}
+
+/**
+ * @brief initialize the fuel gauge
+ *
+ * @return 0 for success
+ */
+static int sbs_gauge_init(const struct device *dev)
+{
+	struct sbs_gauge_config *cfg;
+
+	cfg = (struct sbs_gauge_config *)dev->config;
+
+	if (!device_is_ready(cfg->i2c_dev)) {
+		LOG_ERR("%s device is not ready", cfg->i2c_dev->name);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api sbs_gauge_driver_api = {
+	.sample_fetch = sbs_gauge_sample_fetch,
+	.channel_get = sbs_gauge_channel_get,
+};
+
+#define SBS_GAUGE_INIT(index) \
+	static struct sbs_gauge_data sbs_gauge_driver_##index; \
+\
+	static const struct sbs_gauge_config sbs_gauge_config_##index = { \
+		.i2c_dev = DEVICE_DT_GET(DT_INST_BUS(index)), \
+		.i2c_addr = DT_INST_REG_ADDR(index), \
+	}; \
+\
+	DEVICE_DT_INST_DEFINE(index, \
+			    &sbs_gauge_init, \
+			    device_pm_control_nop, \
```

Comment:
`device_pm_control_nop` was deprecated in #34682

Reply:
thx for the hint, device_pm_control_nop replaced by NULL